src/core/mac/mac.cpp

```diff
@@ -620,6 +634,29 @@ void Mac::SentFrame(bool aAcked)
             }
         }
 
+        if (mReceiveTimer.IsRunning())
+        {
+            mCounters.mTxDataPoll++;
+        }
+        else
+        {
+            mCounters.mTxData++;
+        }
+
+        if (sendFrame.GetAckRequest())
+        {
+            mCounters.mTxAckRequested++;
```

Comment:
This is what I was looking for.  This would be simpler if we moved this to the top of `Mac::SentFrame`.